code/server/controllers/SelectedHikeManager.js

```diff
@@ -150,6 +157,31 @@ class SelectedHikeManager {
 
     return this.updateSelectedHike({ ...selectedHike, endTime: endTime, status: "finished"}, "selectedHikeId", selectedHikeId);
   }
+
+
+async startHike(hikeId, startTime,hikerId) {
```

Comment:
You should add checking that start time is less or equal to now and that the hiker has not another started hike overlapped with the one he want to start in that start time